File: client-android/app/src/main/java/stasis/client_android/scheduling/Schedules.kt
package stasis.client_android.scheduling

import stasis.client_android.lib.model.server.schedules.Schedule
import stasis.client_android.lib.ops.scheduling.ActiveSchedule

data class Schedules(
    val public: List<Schedule>,
    val local: List<Schedule>,
    val configured: List<ActiveSchedule>
) {
    companion object {
        fun emtpy(): Schedules = Schedules(
            public = emptyList(),
            local = emptyList(),
            configured = emptyList(),
        )
    }
}

File: client-ios/app/StasisClient/Scheduling/Schedules.swift
import StasisClientLib

public struct Schedules: Sendable, Equatable {
    public let publicSchedules: [Schedule]
    public let local: [Schedule]
    public let configured: [ActiveSchedule]

    public init(
        publicSchedules: [Schedule] = [],
        local: [Schedule] = [],
        configured: [ActiveSchedule] = []
    ) {
        self.publicSchedules = publicSchedules
        self.local = local
        self.configured = configured
    }

    public static let empty = Schedules()
}
